js/kraken.js

```diff
@@ -528,6 +530,142 @@ module.exports = class kraken extends Exchange {
+            amount = Math.abs (amount);
+        } else {
+            direction = 'in';
+        }
+        let time = this.safeFloat (data, 'time');
+        let timestamp = undefined;
+        let datetime = undefined;
+        if (time !== undefined) {
+            timestamp = parseInt (time * 1000);
+            datetime = this.iso8601 (timestamp);
+        }
+        let fee = {
+            'cost': this.safeFloat (data, 'fee'),
+            'currency': code,
+        };
+        let balanceBefore = undefined;
+        let balanceAfter = this.safeFloat (data, 'balance');
+        return {
+            'info': data,
+            'id': id,
+            'direction': direction,
+            'account': account,
+            'referenceId': referenceId,
+            'referenceAccount': referenceAccount,
+            'type': type,
+            'currency': code,
+            'amount': amount,
+            'balanceBefore': balanceBefore,
+            'balanceAfter': balanceAfter,
+            'timestamp': timestamp,
+            'datetime': datetime,
+            'fee': fee,
+        };
+    }
+
+    async fetchLedger (code = undefined, since = undefined, limit = undefined, params = {}) {
+        // https://www.kraken.com/features/api#get-ledgers-info
+        await this.loadMarkets ();
+        let request = {};
+        let currency = undefined;
+        if (code !== undefined) {
+            currency = this.currency (code);
+            request['asset'] = currency['id'];
+        }
+        if (since !== undefined) {
+            request['start'] = parseInt (since / 1000);
+        }
+        let response = await this.privatePostLedgers (this.extend (request, params));
+        // {  error: [],
+        //   result: { ledger: { 'LPUAIB-TS774-UKHP7X': {   refid: "A2B4HBV-L4MDIE-JU4N3N",
+        //                                                   time:  1520103488.314,
+        //                                                   type: "withdrawal",
+        //                                                 aclass: "currency",
+        //                                                  asset: "XETH",
+        //                                                 amount: "-0.2805800000",
+        //                                                    fee: "0.0050000000",
+        //                                                balance: "0.0000051000"           },
+        let ledger = response['result']['ledger'];
+        let keys = Object.keys (ledger);
+        let data = [];
+        for (let i = 0; i < keys.length; i++) {
+            let key = keys[i];
+            let value = ledger[key];
+            value['id'] = key;
+            data.push (value);
+        }
+        return this.parseLedger (data, currency, since, limit);
+    }
+
+    async fetchLedgerItemsByIds (ids, code = undefined, params = {}) {
+        // https://www.kraken.com/features/api#query-ledgers
+        await this.loadMarkets ();
+        if (Array.isArray (ids)) {
```

Comment:
Let's remove this conditional, because this method will expect an array by specification.
We should also filter by these `ids`,`since`, `limit` and `currency` before returning the result.

Reply:
> It is already implemented in the base class method parseLedger

 Exactly! But we're not passing them to parseLedger anywhere near that particular line in the code i commented on.